webauthn/helpers.js

```diff
@@ -537,28 +537,22 @@ function validateAuthenticatorAssertionResponse(assert) {
 
 function standardSetup(cb) {
     // Setup an automated testing environment if available.
-    let authenticator;
-    promise_test(async t => {
-        try {
-            authenticator = await window.test_driver.add_virtual_authenticator({
-                protocol: "ctap1/u2f",
-                transport: "usb"
-            });
-        } catch (error) {
-            if (error !== "error: Action add_virtual_authenticator not implemented") {
-                throw error;
-            }
-            // The protocol is not available. Continue manually.
-        }
-    }, "Set up the test environment");
-
-    cb();
-
-    promise_test(t => {
-        if (authenticator) {
-            return window.test_driver.remove_virtual_authenticator(authenticator);
+    window.test_driver.add_virtual_authenticator({
+        protocol: "ctap1/u2f",
+        transport: "usb"
+    }).then(authenticator => {
+        cb();
+        // XXX add a subtest to clean up the virtual authenticator since
+        // testharness does not support waiting for promises on cleanup.
```

Comment:
Well `t.add_cleanup` can take a promise if `t` is a promise test. The problem is the lack of a global `add_cleanup`.

Reply:
Yes, and I considered creating a virtual authenticator per subtest but that would bring other problems so I think for now this should be fine.

Reply:
Not sure if you'd like to update this comment, but otherwise LGTM